go/mcap/indexed_message_iterator.go

```diff
@@ -40,6 +40,7 @@ type indexedMessageIterator struct {
 	hasReadSummarySection bool
 
 	compressedChunkAndMessageIndex []byte
+	metadataCallback               func(*Metadata) error
```

Comment:
nit: conceptually/name-wise it's a bit weird that this is a "message iterator", yet it can also yield Metadata records.

Reply:
it can't yield metadata records - it handles them with a callback. But maybe that's still weird? Not sure.